packages/alfa-rules/src/sia-r62/rule.ts

```diff
@@ -235,8 +237,15 @@ function hasNonLinkText(device: Device): Predicate<Element> {
         flattened: true,
       });
 
-      // If we've found text, we're done.
-      if (children.some(and(isText, isVisible(device)))) {
+      // If we've found text with more than whitespaces, we're done.
+      if (
+        children.some(
+          and(
+            isText,
+            and(isVisible(device), (text: Text) => !isWhitespace(text.data))
```

Comment:
```suggestion
            and(isVisible(device), (text) => !isWhitespace(text.data))
```
Type annotation is not needed here because it has already been narrowed by `isText`.